stdlib/public/core/StringCharacterView.swift

```diff
@@ -739,6 +741,12 @@ extension String.CharacterView {
   ///
   /// - Complexity: O(*n*) if the underlying string is bridged from
   ///   Objective-C, where *n* is the length of the string; otherwise, O(1).
+  @available(swift, introduced: 4.0)
+  public subscript(bounds: Range<Index>) -> Substring.CharacterView {
```

Comment:
Right now when the bridging happens, a new `String` gets created from a substring [here](https://github.com/apple/swift/blob/41b00c57ab90c176e9d24098f839b315a2812a50/stdlib/public/SDK/Foundation/String.swift#L63).